slp/modules/rnn.py

```diff
@@ -60,9 +65,11 @@ def __init__(
 
     def forward(self, x, lengths):
         x = self.embed(x)
-        out, last_hidden = self.rnn(x, lengths)
+        out, last_hidden, hidden = self.rnn(x, lengths)
         if self.attention is not None:
-            out = self.attention(
-                out, attention_mask=pad_mask(lengths).to(self.device))
+            out, _ = self.attention(
+                out, attention_mask=pad_mask(lengths, device=self.device))
```

Comment:
@ckarouzos Have you tried using attention after the fix?

Reply:
Checked, working.